src/app/components/core/notifications/add-notification-wizard/add-notification-wizard.component.ts

```diff
@@ -0,0 +1,445 @@
+    } else {
+      this.description.nativeElement.value = '';
+    }
+  }
+
+  onCheckboxClicked(event) {
+    if (event.target.checked) {
+      this.isNotificationEnabled = true;
+    } else {
+      this.isNotificationEnabled = false;
+    }
+    this.payload.enabled = this.isNotificationEnabled;
+  }
+
+  /**
+   * Get edited configuration from view config child page
+   * @param changedConfig changed configuration of a selected plugin
+   */
+  getPluginChangedConfig(changedConfig: any, pluginConfigurationData: any, pageId: number) {
+    const defaultConfig = map(pluginConfigurationData.value[0], (v, key) => ({ key, ...v }));
+    // make a copy of matched config items having changed values
+    const matchedConfig = defaultConfig.filter(e1 => {
+      return changedConfig.some(e2 => {
+        return e1.key === e2.key;
+      });
+    });
+
+    // make a deep clone copy of matchedConfig array to remove extra keys(not required in payload)
+    const matchedConfigCopy = cloneDeep(matchedConfig);
+    /**
+     * merge new configuration with old configuration,
+     * where value key hold changed data in config object
+    */
+    matchedConfigCopy.forEach(e => {
+      changedConfig.forEach(c => {
+        if (e.key === c.key) {
+          e.value = c.value.toString();
+        }
+      });
+    });
+
+    // final array to hold changed configuration
+    const finalConfig = [];
+    matchedConfigCopy.forEach(item => {
+      finalConfig.push({
+        [item.key]: item.type === 'JSON' ? JSON.parse(item.value) : item.value
+      });
+    });
+
+    if (pageId === 3) {
+      this.rulePluginChangedConfig = reduce(finalConfig, function (memo, current) { return assign(memo, current); }, {});
+    } else if (pageId === 5) {
+      this.deliveryPluginChangedConfig = reduce(finalConfig, function (memo, current) { return assign(memo, current); }, {});
+    }
+  }
+
+  getNotificationTypeList() {
+    this.notificationService.getNotificationTypeList()
+      .subscribe(
+        (data: []) => {
+          this.notificationTypeList = data['notification_type'];
+          this.notificationForm.controls['type'].setValue(this.notificationTypeList[0]);
+        },
+        (error) => {
+          if (error.status === 0) {
+            console.log('service down ', error);
+          } else {
+            this.alertService.error(error.statusText);
+          }
+        });
+  }
+
+  /**
+   * Method to add service
+   * @param payload  to pass in request
+   * @param nxtButton button to go next
+   * @param previousButton button to go previous
+   */
+  public addNotificationInstance(payload: any) {
+    /** request started */
+    this.ngProgress.start();
+    this.notificationService.addNotificationInstance(payload)
+      .subscribe(
+        () => {
+          /** request done */
+          this.ngProgress.done();
+          this.alertService.success('Notification service added successfully.', true);
```

Comment:
what is the API response here. can not we use that as is. if that is not that good, better to say `'Notification instance <name> added successfully.`